x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/async/AsyncTaskServiceTests.java

```diff
@@ -179,6 +182,118 @@ public void testAutoCreateIndex() throws Exception {
         }
     }
 
+    public void testUpdateDeletedResponse() throws Exception {
+        AsyncExecutionId id = new AsyncExecutionId("0", new TaskId("N/A", 0));
+        {
+            AsyncSearchResponse resp = new AsyncSearchResponse(id.getEncoded(), true, true, 0L, 0L);
+            PlainActionFuture<IndexResponse> future = PlainActionFuture.newFuture();
+            indexService.createResponse(id.getDocId(), Collections.emptyMap(), resp, future);
+            future.get();
+        }
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
should we have CountDownLatch with more than > 1 counts, otherwise we may not have even a single proper update.  May be can put `latch.countDown();` inside iterations for loop? WDYT?